packages/web/src/components/track/GiantTrackTile.tsx

```diff
@@ -526,12 +535,23 @@ export const GiantTrackTile = ({
           </div>
 
           <div className={cn(styles.playSection, fadeIn)}>
-            <PlayPauseButton
-              doesUserHaveAccess={doesUserHaveAccess}
-              playing={playing}
-              onPlay={onPlay}
-              trackId={trackId}
-            />
+            {doesUserHaveAccess ? (
```

Comment:
This logic is slightly confusing.. but I think I parsed it correctly

- Both normal play button and preview play button if user owns the track (shouldn't preview only appear for premium content?)
- Only the normal play button if user has access
- Only the preview play button if user does not have access _and_ it's purchase gated
- Nothing if the user doesn't have access and it's not purchase gated

It looks like we should always render the play/pause button but disabled if you don't have access for other gate types, so we might be missing a condition on this, maybe `doesUserHaveAccess || !isPurchasedGated`?

Also maybe we rename `doesUserHaveAccess` to `disabled` in PlayPauseButton and just pass through the disabled state directly, as it seems like that's all it's used for inside the PlayPauseButton component? wdyt?

Lastly minor nit but anything to make this logic easier to understand (comments or refactors/renames?) would probably be helpful in the future

Reply:
I agree that it's confusing. I think you got the analysis right. And based on that I made a mistake on one of the cases (showing a disabled play button instead of no button for non-purchase gated tracks).I updated it to:
* Calculate `showPlay` and `showPreview` explicitly outside of the rendering block. Each has a comment explaining what the conditions are
* Use `disabled` in the button components instead of passing `doesUserHaveAccess`, and then deriving the former from the latter as needed.